src/vsr/replica.zig

```diff
@@ -2094,19 +2093,15 @@ pub fn ReplicaType(
 
             const op_checkpoint_max =
                 DVCQuorum.op_checkpoint_max(self.do_view_change_from_all_replicas);
-            if (op_checkpoint_max > self.op_checkpoint() and op_head > self.op_prepare_max()) {
-                // When:
-                // 1. the cluster is at a checkpoint ahead of the local checkpoint,
-                // 2. AND the quorum's head op is part of a future checkpoint,
-                //    abdicate as primary, jumping to a new view.
-                //
-                // - If 1 and ¬2, then the ops in the next checkpoint are uncommitted and our
-                //   current checkpoint is sufficient.
-                // - If 2 and ¬1, then the ops past the checkpoint's op-prepare-max must be
-                //   uncommitted because no replica in the DVC quorum even prepared them.
-                //   (They are present anyway because the origin of the DVC is reusing its SV
-                //   headers as a DVC since it has not completed repair.)
-                //
+            const commit_max = @max(
+                self.commit_max,
+                DVCQuorum.commit_max(self.do_view_change_from_all_replicas),
```

Comment:
explain why we can't rely on our commit_max being a part of DVCQUorum (and using our commit_min, imo, makes this a bit clearer)

Reply:
Added some assertions and comments while computing commit_max for potential primary (including `assert(self.commit_min <= DVCQuorum.commit_max(self.do_view_change_from_all_replicas));` as you suggested in https://github.com/tigerbeetle/tigerbeetle/pull/2236#discussion_r1738868671): https://github.com/tigerbeetle/tigerbeetle/pull/2236/commits/161ffd0652c247f3d5c4cbe651123a7b03e56251

Thoughts?

Reply:
Ah, I certainly needed to add this _wrong_ assert to understand this better! It turns out commit_min _can_ be greater than commit_max derived from the DVCQuorum. Relaxed our asserts in https://github.com/tigerbeetle/tigerbeetle/pull/2236/commits/fbfcbffbe0e6e19672361a59b5718d6f9a6180b7, hopefully it is clearer now!